.github/workflows/lint.yml

```diff
@@ -0,0 +1,107 @@
+name: Lint
+
+on:
+  pull_request:
+
+jobs:
+  # Determine what the js changes are
+  js-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      packages: ${{ steps.filter.outputs.changes }}
+    steps:
+      # Use paths-filter to determine which paths have changes
+      # https://github.com/dorny/paths-filter
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # Format of key: value. When there is a change detected in value, the
+          # key will be returned in the output.
+          filters: |
+            packages/dashboard: packages/dashboard/**
+            packages/dashboard-v2: packages/dashboard-v2/**
+            packages/dnslink-api: packages/dnslink-api/**
+            packages/handshake-api: packages/handshake-api/**
+            packages/health-check: packages/health-check/**
+            packages/website: packages/website/**
+
+  # Determine what the dockerfile changes are
+  dockerfile-changes:
```

Comment:
shouldn't we have just one "changes" job that contains multiple filters? Sounds a bit easier to maintain

example: https://github.com/GoogleChrome/web.dev/blob/e1f0c28964e99ce6a996c1e3fd3ee1985a7a04f6/.github/workflows/lint-and-test-workflow.yml#L33-L48

Reply:
The reason we need the different change blocks is because the linters are expecting different values. If the change block was all one we would need to add more complicated filtering to the action calls.

The JS linters need the package paths for the working directory.

The hadolint needs the dockerfile path.

The python check is the only one that we would be able to mimic that functionality.

If we combined them all we would get an output like this
```
['packages/health-check', 'docker/nginx//Dockerfile', 'py']
```

So we would then need to filter the filter results to properly pass the right arguments to the callers. 

Keeping the changes separate is much simpler.